extension/extension.go

```diff
@@ -51,6 +51,15 @@ type ConfigWatcher interface {
 	NotifyConfig(ctx context.Context, conf *confmap.Conf) error
 }
 
+// ModuleInfo describes the go module for each component.
+type ModuleInfo struct {
+	Receiver  map[component.Type]string
```

Comment:
Should this have more structure? For example, should we use https://pkg.go.dev/golang.org/x/mod@v0.20.0/module#Version (or something similar) instead? 

I am a bit hesitant about using `module.Version` directly since that module has a 0.x version, but something similar may make sense

Reply:
I'm not opposed to structuring it more but it wasn't clear to me either what that structure should be. I think it's worth noting that the module string is something of a standard in its own right, since its interpretable by go tooling.

Reply:
> I'm not opposed to structuring it more but it wasn't clear to me either what that structure should be

Hm, I guess a light wrapper around the `module.Version` I linked above makes sense. 

But thinking about it more I remembered open-telemetry/opentelemetry-collector-contrib/issues/31136, where we put `int` as the type for a stepping number, that is commonly represented as a string, and then found out that there are edge cases where it's not representable as an int. 

I guess a wrapper would make this situation manageable, but the lesson I learned there is that using a string may be better when you don't control the evolution of what you are representing.

This is all to say: if you feel like a string works, let's go with a string :)